index.js

```diff
@@ -82,35 +88,17 @@ function buildProcessTree (parentPid, tree, pidsToProcess, cb) {
             if (Object.keys(pidsToProcess).length == 0) {
                 cb();
             }
-            return
-        }
-
-        pids = [];
-        pid = '';
-        for (i = 0; i < allData.length; i++) {
-            if (allData[i] == '\n') {
-                pids.push(parseInt(pid, 10));
-                pid = '';
-                continue;
-            }
-            if (allData[i] != ' ') {
-                pid += allData[i];
-            }
+            return;
         }
 
-        pids.forEach(function (pid) {
-            tree[parentPid].push(pid)
-            tree[pid] = [];
-            pidsToProcess[pid] = 1;
-            buildProcessTree(pid, tree, pidsToProcess, cb);
+        allData.match(/\d+/g).forEach(function (pid) {
+          pid = parseInt(pid, 10);
```

Comment:
As far as I can see, the `ps` output was composed of digits, spaces and line feeds. The old code ignored spaces  that were used as left padding, stored digits in a buffer, and upon reaching a line feed token the buffer would be flushed pushing its contents to the PIDs array. The old code also called `parseInt` for each PID, which discards everything after a non-leading/trailing-whitespace non-digit character. In this sense, the `/\d+/` is equivalent to the old parser.